Dockerfile.local

```diff
@@ -39,63 +41,82 @@ RUN git clone -b v${MUT_VERSION} --depth 1 https://github.com/mongodb/mut.git \
+RUN chown -Rv docsworker-xlarge ${WORK_DIRECTORY}
+USER docsworker-xlarge
 
 # install snooty frontend and docs-tools
 RUN git clone -b v${SNOOTY_FRONTEND_VERSION} --depth 1 https://github.com/mongodb/snooty.git       \
     && cd snooty                                                                                   \
     && npm ci --legacy-peer-deps --omit=dev                                                        
 
+RUN curl https://raw.githubusercontent.com/mongodb/docs-worker-pool/meta/makefiles/shared.mk -o shared.mk
+
+
+RUN git clone -b @dop/redoc-cli@${REDOC_CLI_VERSION} --depth 1 https://github.com/mongodb-forks/redoc.git redoc \
+    # Install dependencies for Redoc CLI
+    && cd redoc/ \
+    && npm ci --prefix cli/ --omit=dev
+    
+FROM initial as persistence
+
 RUN mkdir -p modules/persistence && chmod 755 modules/persistence
 COPY modules/persistence/package*.json ./modules/persistence/
 RUN cd ./modules/persistence \
     && npm ci --legacy-peer-deps 
-
-RUN  mkdir -p modules/oas-page-builder && chmod 755 modules/oas-page-builder
-COPY modules/oas-page-builder/package*.json ./modules/oas-page-builder/
-RUN cd ./modules/oas-page-builder \
-    && npm ci --legacy-peer-deps 
-
-# Root project build
-COPY package*.json ./
-RUN npm ci --legacy-peer-deps
 # Build persistence module
 
-COPY --chown=docsworker modules/persistence/tsconfig*.json ./modules/persistence
-COPY --chown=docsworker modules/persistence/src ./modules/persistence/src/
-COPY --chown=docsworker modules/persistence/index.ts ./modules/persistence
+COPY --chown=docsworker-xlarge modules/persistence/tsconfig*.json ./modules/persistence
+COPY --chown=docsworker-xlarge modules/persistence/src ./modules/persistence/src/
+COPY --chown=docsworker-xlarge modules/persistence/index.ts ./modules/persistence
 
 RUN cd ./modules/persistence \
     && npm run build
 
+FROM initial as oas
+
+RUN  mkdir -p modules/oas-page-builder && chmod 755 modules/oas-page-builder
+COPY modules/oas-page-builder/package*.json ./modules/oas-page-builder/
+RUN cd ./modules/oas-page-builder \
+    && npm ci --legacy-peer-deps 
 # Build modules
 # OAS Page Builder
-COPY --chown=docsworker modules/oas-page-builder/tsconfig*.json ./modules/oas-page-builder
-COPY --chown=docsworker modules/oas-page-builder/src ./modules/oas-page-builder/src/
-COPY --chown=docsworker modules/oas-page-builder/index.ts ./modules/oas-page-builder
+COPY --chown=docsworker-xlarge modules/oas-page-builder/tsconfig*.json ./modules/oas-page-builder
+COPY --chown=docsworker-xlarge modules/oas-page-builder/src ./modules/oas-page-builder/src/
+COPY --chown=docsworker-xlarge modules/oas-page-builder/index.ts ./modules/oas-page-builder
 
 RUN cd ./modules/oas-page-builder \
     && npm run build
 
+COPY --chown=docsworker-xlarge ./modules/oas-page-builder/dist/ ./modules/oas-page-builder
+
+FROM oas as root
+
+COPY --from=persistence --chown=docsworker-xlarge ${WORK_DIRECTORY}/modules/persistence/ ./modules/persistence
```

Comment:
@branberry Just making a last-minute check because of our new `COPY` knowledge. Is it ok that we have six `COPY` statements without using `./` at the beginning of the first path? Also, there's a seventh (in a way) on line 99 because `WORK_DIRECTORY` does not begin with a `.`

Reply:
For what it's worth, I think the one on line 99 is pretty safe since `--from` would explicitly ensure we're copying from the intended source. If we wanted to play it safe, we can try using `--from` on the other copy commands. We do so on other Dockerfiles, but not sure if the use case is different?

Reply:
Oh wait, nvm, I think `--from` would only be applicable for previous build steps, not if you're copying from local source. Feel free to ignore!